Release notes for NPE fix in API logging in 1.7.1 (#4539)
dapr/dapr: docs/release_notes/v1.7.1.md

> This hotfix release addresses a change made in how Actor Timers are validated during creation. Specifically, `DueTime` can be set to 0 to have a reminder fire immediately.

Reviewer: ```suggestion
This hotfix release addresses a change made in how actor timers are validated during creation. Specifically, `DueTime` can be set to 0 to have a reminder fire immediately.
```

> Additionally, this hotfix addresses an issue in the Apache Pulsar Pubsub component preventing the use of namespaces when publishing to topics.

Reviewer: ```suggestion
Additionally, this hotfix addresses an issue in the Apache Pulsar pubsub component preventing the use of namespaces when publishing to topics.
```

> Lastly, there is a fix to avoid a nil dereference crash for API logging in gRPC API.

Reviewer: ```suggestion
Lastly, there is a fix to avoid a nil dereference crash on the sidecar when using API logging feature in gRPC API.
```

#### Solution

Revert the code to pre-1.6 behavior which allows `DueTimes` to be in the past as that still equates to firing immediately.

### Panic: nil pointer dereference in API Logging

#### Problem

> When enabling API logging, a nil reference can happen and crash the sidecar.

Reviewer: ```suggestion
When enabling the API logging preview feature, a nil reference can occur and crash the sidecar.
```

#### Root cause

> Logic for API logging in gRPC assumed gRPC `info` object was always not nil, but user reported it could be nil when invoking an Actor as per issue [#4527](https://github.com/dapr/dapr/issues/4527).

Reviewer: ```suggestion
Logic for API logging in gRPC assumed gRPC `info` object was always not nil, however it has been reported that it could be nil when invoking an actor as per issue [#4527](https://github.com/dapr/dapr/issues/4527).
```

#### Solution

> Add a nil check for API logging in gRPC API.

Reviewer: ```suggestion
Add a nil check for API logging preview feature in gRPC API.
```
